dashboard/modules/serve/serve_agent.py

```diff
@@ -58,6 +58,8 @@ async def get_all_deployments(self, req: Request) -> Response:
         else:
             try:
                 config = await controller.get_app_config.remote()
+                if config is None:
```

Comment:
could you explain why we need to make`get_app_config` returning None instead of  `ServeApplicationSchema.get_empty_schema_dict()` ?

Reply:
Because `ServeApplicationSchema.get_empty_schema_dict()` can't actually parse correctly as `ServeApplicationSchema`. It has `import_path` set to the empty string. 

This is because back when there was only a single application, `ServeApplicationSchema.get_empty_schema_dict()` was used as the response for `GET /api/serve/deployments/` for when Serve hasn't started, or when no application has been deployed.

Reply:
Another option is to track app configs as part of application state manager, that way we can get the app config even for applications that are in DELETING state. It didn't pop out to me as that important, so I didn't go down that route.